### PR TITLE
Allow custom script execution on installation

### PR DESCRIPTION
**Pull request by mgerhardy**
Should fix issues
- https://github.com/nextcloud/docker/issues/820
- https://github.com/nextcloud/docker/issues/1203

This allows a custom script to be mounted as docker or kubernetes volume to be executed. This can be used to execute all kind of occ commands.

Alternative and more customizable to the PR https://github.com/nextcloud/docker/pull/1259

### CODE REVIEW

**Comment by mgerhardy** [comment]
one could e.g. do

```bash
#!/bin/sh

occ() {
    php /var/www/html/occ $@
}

# loginname, display name, email, password
createuser() {
    export OC_PASS="$4"
    occ user:add --password-from-env --display-name="$2" "$1"
    export OC_PASS=""
    occ user:setting mgerhardy email "$3"
}

for app in contacts calendar; do
    occ app:install -n $app
done

createuser user1 "User 1" "user@domain.com" "secrets-mounted-as-docker-or-k8s-secret"
```

mount this to `/tmp/configure.sh` and set `NEXTCLOUD_CUSTOM_INSTALL_SCRIPT` to `/tmp/configure.sh`

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -168,6 +168,10 @@ if expr "$1" : "apache" 1>/dev/null || [ "$1" = "php-fpm" ] || [ "${NEXTCLOUD_UP
                             NC_TRUSTED_DOMAIN_IDX=$(($NC_TRUSTED_DOMAIN_IDX+1))
                         done
                     fi
+
+                    if [ -n "${NEXTCLOUD_CUSTOM_INSTALL_SCRIPT+x}" ]; then
+                        run_as "/bin/sh ${NEXTCLOUD_CUSTOM_INSTALL_SCRIPT}"
```
Comment (J0WI) [inline docker-entrypoint.sh:173]: I'm fine with this but are there any common use cases where you don't want to `run_as` limited user?

Comment (Loxeno) [reply docker-entrypoint.sh:173]: The only use case I see may be adding distrib packages, but I think that building a custom image is better here.

**Comment by J0WI** [comment]
@mgerhardy would you like to rebase this?

**Comment by Loxeno** [comment]
Hi,

Any news @mgerhardy ?

**Comment by Loxeno** [comment]
Hi,

@mgerhardy, do you need any help ?

Thank ! ;)

**Pull request closed without merging**